lib/fog/linode/requests/compute/avail_kernels.rb

```diff
@@ -5,27 +5,30 @@ class Real
         # Get available kernels
         #
         # ==== Parameters
-        # * kernelId<~Integer>: id to limit results to
+        # * options<~Hash>
+        #   * isXen<~Boolean> Show or hide Xen compatible kernels
+        #   * isKVM<~Boolean> Show or hide KVM compatible kernels
         #
         # ==== Returns
         # * response<~Excon::Response>:
         #   * body<~Array>:
         # TODO: docs
-        def avail_kernels(kernel_id=nil)
-          options = {}
-          if kernel_id
-            options.merge!(:kernelId => kernel_id)
-          end
+        def avail_kernels(options={})
+          # avail.kernels used to accept a kernelId parameter (now removed)
+          raise Fog::Errors::Error.new('avail_kernels no longer accepts a kernelId parameter') unless !options || options.is_a?(Hash)
```

Comment:
@plribeiro3000 `options = nil` and `options = {}` are both allowed. Both will return all the available kernels. The exception is raised if `options` is not `nil` and is not a `Hash`. This is to catch cases where callers may be using the old `kernelId` method signature.